io/disk/port_bounce.py

```diff
@@ -0,0 +1,314 @@
+        """
+        test parameters
+        """
+        self.switch_name = self.params.get("switch_name", '*', default=None)
+        self.userid = self.params.get("userid", '*', default=None)
+        self.password = self.params.get("password", '*', default=None)
+        self.pci_adrs = self.params.get("pci_device", default=None).split(",")
+        self.sbt = int(self.params.get("sbt", '*', default=10))
+        self.lbt = int(self.params.get("lbt", '*', default=250))
+        self.count = int(self.params.get("count", '*', default="2"))
+        self.prompt = ">"
+        self.verify_sleep_time = 20
+        self.port_ids = []
+        self.dic = {}
+        for pci_id in self.pci_adrs:
+            port_id = self.get_switch_port(pci_id)
+            self.port_ids.append(port_id)
+            self.dic[port_id] = pci_id
+
+    def fc_login(self, switch_ip, username, password):
+        '''
+        telnet Login method for remote fc switch
+        '''
+        self.tnc = telnetlib.Telnet(switch_ip)
+        self.tnc.read_until('login: ')
+        self.tnc.write(username + '\n')
+        self.tnc.read_until('assword: ')
+        self.tnc.write(password + '\n')
+        ret = self.tnc.read_until(self.prompt)
+        assert self.prompt in ret
+
+    def _send_only_result(self, command, response):
+        output = response.splitlines()
+        if command in output[0]:
+            output.pop(0)
+        output.pop()
+        output = [element.lstrip()+'\n' for element in output]
+        response = ''.join(output)
+        response = response.strip()
+        return ''.join(response)
+
+    def fc_run_command(self, command, timeout=300):
+        '''
+        Telnet Run command method for running commands on fc switch
+        '''
+        self.log.info("Running the command on fc switch %s", command)
+        if not hasattr(self, 'tnc'):
+            self.fail("telnet connection to the fc switch not yet done")
+        self.tnc.write(command + '\n')
+        response = self.tnc.read_until(self.prompt)
+        return self._send_only_result(command, response)
+
+    def test(self):
+        '''
+        Test method
+        '''
+        self.failure_list = {}
+        self.port_bounce()
+        if self.failure_list:
+            self.fail("failed ports, details: %s" % self.failure_list)
+
+    def port_bounce(self):
+        '''
+        defins and test for different scenarios
+        '''
+        test_ports = []
+        self.log.info("short port bounce for individual ports:")
+        for port in self.port_ids:
+            test_ports = []
+            test_ports.append(port)
+            for _ in range(self.count):
+                self.porttoggle(test_ports, self.sbt)
+
+        self.log.info("Long port bounce for individual ports:")
+        for port in self.port_ids:
+            test_ports = []
+            test_ports.append(port)
+            for _ in range(self.count):
+                self.porttoggle(test_ports, self.lbt)
```

Comment:
`test_ports` is initialised multiple times, take care of that.
Also, `test_ports` can be found once, and used the subsequent times.
And the 2 sequences do the same thing except for the port toggle time, so they can be made into a function and called twice.